packages/jest-mock/src/index.js

```diff
@@ -173,31 +173,36 @@ function matchArity(fn: any, length: number): any {
   return mockConstructor;
 }
 
-function isA(typeName: string, value: any): boolean {
-  return Object.prototype.toString.apply(value) === '[object ' + typeName + ']';
+function getObjectType(value: any): string {
+  return Object.prototype.toString.apply(value).slice(8, -1);
```

Comment:
Do you have any perf benchmarks showing that this slice is not slowing down the whole caching effort?

Reply:
`slice(8, -1)` works 4x faster than concatanation `'[object ' + typeName + ']'`.

Source of my test.
```js
const map1 = new Map();
const map2 = new Map();
let start;

function isA(typeName, value) {
  return Object.prototype.toString.apply(value) === '[object ' + typeName + ']';
}

function getObjectType(value) {
  return Object.prototype.toString.apply(value).slice(8, -1);
}

console.log('isA');
console.log('=======================');

start = process.hrtime()[1];
const compare1 = isA('Map', map1);
console.log(process.hrtime()[1] - start, compare1);


console.log('');
console.log('getObjectType');
console.log('=======================');
start = process.hrtime()[1];
const compare2 = getObjectType(map1) === 'Map';
console.log(process.hrtime()[1] - start, compare2);
```

Result of my tests (nano seconds):
```
isA
=======================
208620 true

getObjectType
=======================
57268 true
```

Tested on Node v10.10.0, MacOs.

Reply:
This is single execution without a warmup, hard to tell if it's representative. Can your run it on jsperf? Something like here https://twitter.com/bmeurer/status/867277137564360704?s=21

Reply:
The gain in jest's case is probably from doing the `toString` once instead of up to 12 times.

I doubt `slice` or not makes much of a difference. I don't think this is super hot code anyways - it's only used when generating mocks

Reply:
Ok. full test.
Test compares original `getType`(with many isA calls) and `getType` from this PR with chaching.

https://jsperf.com/object-prototype-tostring-caching

Reply:
here's the result:

![](http://dp.hanlon.io/76c1f97caa9c/Image%202018-10-14%20at%209.34.12%20PM.png)

fwiw, performance aside, I think this code is easier to read and maintain so I'm +1